### PR TITLE
Remove obsolete RegistrationCatalogCollector and its fall-out

### PR DESCRIPTION
**Pull request by maartenba**
Please review @skofman1 @xavierdecoster

### CODE REVIEW

File: src/Catalog/RegistrationCatalogCollector.cs [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
-using System;
-using System.Collections.Generic;
-using System.Net.Http;
-using System.Threading;
-using System.Threading.Tasks;
-using VDS.RDF;
-
-namespace NuGet.Services.Metadata.Catalog
-{
-    public class RegistrationCatalogCollector : SortingGraphCollector
```
Comment (skofman1) [inline src/Catalog/RegistrationCatalogCollector.cs]: but this guy is used by Catalog2Registration..

Comment (maartenba) [inline src/Catalog/RegistrationCatalogCollector.cs]: We're using the RegistrationMaker in there, this one is not used anymore (hence the PR)

Comment (skofman1) [inline src/Catalog/RegistrationCatalogCollector.cs]: My bad :) :shipit:

Comment (maartenba) [inline src/Catalog/RegistrationCatalogCollector.cs]: At least this clears up confusion :)

**Pull request merged**